src/features/device-state/routes/state-patch-v3.ts

```diff
@@ -334,6 +334,26 @@ export const statePatchV3: RequestHandler = async (req, res) => {
 					});
 				}
 
+				if (Object.keys(metricsBody).length > 0) {
+					updateFns.push(async (resinApiTx) => {
+						if (latestDeviceMetricsRecord == null) {
+							await resinApiTx.post({
+								resource: 'device_metrics_record',
+								body: {
+									...metricsBody,
+									...{ is_reported_by__device: device.id },
+								},
+							});
+						} else {
+							await resinApiTx.patch({
+								resource: 'device_metrics_record',
+								id: { is_reported_by__device: device.id },
+								body: metricsBody,
```

Comment:
How about adding v both here and in the v2 patch?
```suggestion
								options: {
									$filter: {
										$not: metricsBody
									}
								},
								body: metricsBody,
```

Reply:
Yeah, this avoids an unnecessary postgres write in the case there are no changes